.ci/error-prone-check.groovy

```diff
@@ -0,0 +1,342 @@
+            " (compile | test-compile) [-g | --generate-suppression]"
+
+    final int exitCode
+    if (profiles.contains(argument)) {
+        if (flag != null && flag != "-g" && flag != "--generate-suppression") {
+            final String exceptionMessage = "\nUnexpected flag: ${flag}" + usageString
+            throw new IllegalArgumentException(exceptionMessage)
+        }
+        exitCode = checkErrorProneReport(argument, flag)
+    }
+    else {
+        final String exceptionMessage = "\nUnexpected argument: ${argument}" + usageString
+        throw new IllegalArgumentException(exceptionMessage)
+    }
+    return exitCode
+}
+
+/**
+ * Check the generated error prone report. Parse the errors and compare them with the suppressed
+ * errors.
+ *
+ * @param profile the error prone profile to execute
+ * @param flag command line argument flag to determine output format
+ * @return {@code 0} if error prone report is as expected, {@code 1} otherwise
+ */
+private static int checkErrorProneReport(String profile, String flag) {
+    final XmlParser xmlParser = new XmlParser()
+    final String suppressedErrorsFileUri =
+            ".${SEPARATOR}.ci${SEPARATOR}" +
+                    "error-prone-suppressions${SEPARATOR}${profile}-phase-suppressions.xml"
+    final List<String> errorProneErrors = getErrorProneErrors(profile)
+    Set<ErrorProneError> errors = Collections.emptySet()
+    if (!errorProneErrors.isEmpty()) {
+        errors = getErrorFromText(errorProneErrors)
+    }
+    final File suppressionFile = new File(suppressedErrorsFileUri)
+    Set<ErrorProneError> suppressedErrors = Collections.emptySet()
+    if (suppressionFile.exists()) {
+        final Node suppressedErrorsNode = xmlParser.parse(suppressedErrorsFileUri)
+        suppressedErrors = getSuppressedErrors(suppressedErrorsNode)
+    }
+    return compareErrors(errors, suppressedErrors, flag)
+}
+
+/**
+ * Generates the error prone report and filters out the errors.
+ *
+ * @param profile the error prone profile to execute
+ * @return A set of errors
+ */
+private static List<String> getErrorProneErrors(String profile) {
+    final List<String> errorProneErrors = [] as ArrayList
+    final String command = "mvn -e --no-transfer-progress clean" +
+            " ${profile} -Perror-prone-${profile}"
+    final Process process = command.execute()
+    process.in.eachLine { line ->
+        if (line.startsWith("[ERROR]")) {
+            errorProneErrors.add(line)
+        }
+        println(line)
+    }
+    process.waitFor()
+    return errorProneErrors
+}
+
+/**
+ * Get a set of {@link ErrorProneError} from text.
+ *
+ * @param errorsText errors in text format
+ * @return A set of errors
+ */
+private static Set<ErrorProneError> getErrorFromText(List<String> errorsText) {
+    final Set<ErrorProneError> errors = new HashSet<>()
+    final Pattern errorExtractingPattern = Pattern
+            .compile(".*/(.*\\.java):\\[(\\d+).*\\[(\\w+)\\](.*)");
```

Comment:
Sad that we have to do this, there has been no action at https://github.com/google/error-prone/issues/444.

Reply:
I added a comment requesting for support, let's hope they support this in future releases.